scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import java.io.FileInputStream
+import org.oppia.android.app.model.FilenameChecks
+import org.oppia.android.app.model.FilenameCheck
+import org.oppia.android.app.model.FileContentChecks
+import org.oppia.android.app.model.FileContentCheck
+import org.oppia.android.scripts.ScriptResultConstants
+
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+
+      val repoPath = args[0] + "/"
+
+      val allowedDirectories: MutableList<String> = ArrayList()
+
+      for (layerIndex in 1 until args.size) {
+        allowedDirectories.add(args[layerIndex])
+      }
+
+      val searchFiles = collectSearchFiles(repoPath, allowedDirectories)
+
+      var scriptFailedFlag = false
+
+      getFilenameChecks().forEach {
+        if (checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameRegexString = it.getProhibitedFilenameRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
```

Comment:
It's a bit more idiomatic to avoid vars in general, but it's not as easy here. ``any`` is an immediate go-to for cases like this (in place of ``forEach``), but it will short-circuit rather than find all failures. We could instead use ``fold``:

```kotlin
val hasFilenameCheckFailure = getFilenameChecks().fold(initial = false) { isFailing, check ->
  // Run check regardless of current failing state for more comprehensive error reporting.
  val checkFailed = checkProhibitedNamePattern(repoPath, searchFiles, check)
  return@fold isFailing || checkFailed
}
```

Ditto for file contents, then you can fail the script conditionally based on whether either flag is true.

Reply:
done